### ISSUE TITLE
Pi Zero relocation adapter boards - video artefacts - pixel 'sparkles' on edges

### ISSUE THREAD

**Issue by solarmon** [c0pperdragon/Amiga-Digital-Video#40]
Hi,

I'm testing out several Pi Zero relocation adapters - there are three types: A, B and C:

Type A is just flips it over:

![image](https://user-images.githubusercontent.com/46369787/110094372-a6fa0600-7d93-11eb-8ba7-74fe6980b84f.png)

Type B shifts it sideways and flips it:

![image](https://user-images.githubusercontent.com/46369787/110094425-b8dba900-7d93-11eb-9f1c-503fe59f4e15.png)

Type C shifts it sideways, flips it and rotate it:

![image](https://user-images.githubusercontent.com/46369787/110094449-bf6a2080-7d93-11eb-9185-0fb3c4a4468b.png)

Type A works fine without any issues.

However, type B and C has video artefacts where on certain edges there are pixel 'sparkles' artefacts.

I'm aware that this is due to the increased signal traces, but is there anything that could be done to improve it?

Thank you.

**Comment by kipper2k** [comment]
shorter tracks definitely increase performance, if you increase track lengths then you should ensure that decoupling caps are as close to IC's as possible. Also extending the board over other areas of the Amiga motherboard can also pickup unwanted noise from the motherboard so you may have to reduce board length and location to reduce the noise

**Comment by solarmon** [comment]
@kipper2k 

Regarding your comment about caps being close to IC's as possible. These are just relocation boards for the Pi connector. I'm not changing anything with the RGBtoHDMI boards.

The artefacts only appear at 'edges' - where an area of colour changes to another. It's hard for me to capture that, but you can see the white pixel in this (poor quality) picture. These white pixels appear and disappear randomly on such 'edges':

![image](https://user-images.githubusercontent.com/46369787/110202043-3370fb00-7e5e-11eb-8a5f-6b04e44d70bc.png)

Why would it only appear at these edges?

**Comment by solarmon** [comment]
I'm actually seeing this shimmering/sparkle artefact using a normal RGBtoHDMI setup - i.e. no relocator or anything.

With the normal setup setup, in the Amiga Test Kit screen, I don't see these artefacts.

However, when I load up workbench, the artefacst appear on the right hand side of any black areas next to lighter areas.

For example:

![image](https://user-images.githubusercontent.com/46369787/110205561-92d80680-7e70-11eb-947c-90fb00af38f3.png)

It is always on the right hand side of the contrasting edge, if that is of significance.

I haven't checked the RGBtoHDMI menu options yet. Maybe there are some settings I need to tweak?

**Comment by de-nugan** [comment]
This is the same fault I wrote about here =>  https://github.com/c0pperdragon/Amiga-Digital-Video/issues/11#issuecomment-769056224

I'm also using a slightly longer connection through a very short IDC cable. I tested a longer cable and saw increased sparkles and artifacts so the length does appear to be critical. I haven't been able to test the 'default' setup as I don't have any of the female pin sockets, so I may have to solder the Pi to the board and try that instead.

**Comment by solarmon** [comment]
So I think I have narrowed down the issue a bit. It seems this issue is only specific to this X-Copy disk I'm using to test. Everything else I've tried so far has been rock solid. The X-Copy disk I'm using is created from the xcopy_1992-11_en_1200.adf image (http://jope.fi/xcopy/zip/xcopy_1992-11_en_1200.zip). If somebody is able to test this out and check if they have the same issue. (You need to press F10 to go in to Workbench mode).

**Comment by solarmon** [comment]
> This is the same fault I wrote about here => [#11 (comment)](https://github.com/c0pperdragon/Amiga-Digital-Video/issues/11#issuecomment-769056224)
> 
> I'm also using a slightly longer connection through a very short IDC cable. I tested a longer cable and saw increased sparkles and artifacts so the length does appear to be critical. I haven't been able to shorten the path as I don't have any of the female pin sockets, so I may have to solder the Pi to the board and try that instead.
> 
> My guess is that the increased path length is making it harder for the flipflops to drive the lines high from 0v (black) so we are losing bits resulting in colour sparkles where there should be grey or mid-tone colours. I aint no engineer though..
> 
> Oh and sorry to disappoint you, I've tried just about everything in the menus. I don't think it can be fixed there, likely only in hardware or perhaps firmware.

If this issue with my relocator boards can't be fixed then I still have the option using my custom RGBtoHDMI that I'm designing and which relocates the Pi board too. Hopefully that won't have the same issue, but  I still need to get these custom boards made and tested.

**Comment by IanSB** [comment]
@solarmon 
Try replacing the 74LVC parts with 74VHC parts. These are slower and fixed the sparkly problem on my rev5 which doesn't have the clock filtering components that are fitted on rev6 & later boards which slightly delay the clock signals.
Note they must be 74VHC parts which are 5v tolerant, not 74HC parts which are not 5v tolerant when powered at 3.3v

@de-nugan 
I also suggest you try the above to fix your issue in  https://github.com/c0pperdragon/Amiga-Digital-Video/issues/11

**Comment by solarmon** [comment]
@IanSB 

Thanks for the suggestion for replacement parts. When you say 'these are slower' - do you mean the 74LVC  or the 74VHC part?

I had already got a batch of the 74LVC parts because that was what was suggested in the schematics. ='(

For reference, these are the parts from LCSC:

U1!: https://lcsc.com/product-detail/74-Series_Nexperia-74LVC86APW-118_C6102.html
U2 and U3: https://lcsc.com/product-detail/74-Series_Nexperia-74LVC574APW_C6098.html

Can you confirm these are the suitable replacement.

U1: https://lcsc.com/product-detail/Pre-ordered-Products_ON-Semiconductor-ON-74VHC86MTCX_C903423.html
U2 and U3: https://lcsc.com/product-detail/Flip-Flops_ON-Semiconductor-ON-MC74VHC574DTR2G_C364877.html

However, the U1 74VHC86MTCX part is not in stock on LCSC.

Are you suggesting that U1, U2 and U3 should all use replacement parts?

**Comment by IanSB** [comment]
@solarmon 
>When you say 'these are slower' - do you mean the 74LVC or the 74VHC part?

The VHC parts are slower. Their propagation delay is about double that of the LVC parts and that allows a longer settle time before latching the data bits.

>Are you suggesting that U1, U2 and U3 should all use replacement parts?

Yes although changing U1 (74LVC86) which handles the clock signal has the most effect so try that first if you are changing parts on an existing board.

>Can you confirm these are the suitable replacement.

They look to be the correct family although I used these toshiba parts from RS:
https://uk.rs-online.com/web/p/products/1713480/
https://uk.rs-online.com/web/p/products/1713428/

**Comment by solarmon** [comment]
@IanSB 

Thanks for the parts link from RS. I'm UK based, so that suits me fine.

BTW - these issues (sparkles due to relocation board, and sparkles using a specific version of X-Copy, and Xenon II) happens on a n A500 rev 6A and an A500plus rev 8A.

**Comment by IanSB** [comment]
>Thanks for the parts link from RS. I'm UK based, so that suits me fine.

The VHC86 parts have gone up a lot in price, I paid £0.056 each a few weeks ago, they are now £0.217 each.
The VHC574 parts haven't gone up (yet!)

>BTW - these issues (sparkles due to relocation board, and sparkles using a specific version of X-Copy, and Xenon II) happens on a n A500 rev 6A and an A500plus rev 8A.

Can't guarantee they will fix your issue but they did fix mine.

**Comment by de-nugan** [comment]
To achieve the shortest possible data path I've just soldered the Pi to the adapter board and this appears to have fixed the sparkle problem. In WB3.1 the sparkles and glitches are now gone, and the screen geometry is now absolutely bang-on pixel perfect straight out of the box. 

This is on my Rev 5 (Assy 312513) A500. The parts used were Nexperia 74LVC574APW.

I'll be building more boards for some other Amigas, do you know if the VHC tip applies to any of the A/B2000 revisions?

**Comment by matthiesenj** [comment]
For HC parts, you should use VHCT or AHCT, which have TTL-compatible inputs.
The point about propagation delay is probably relevant, since the original c0pperdragon design relies on the propagation delay of two gates to generate the 14MHz flipflop clock.

**Comment by solarmon** [comment]
@matthiesenj 

I'm confused now - @IanSB mentioned '74VHC' parts and now you are mentioning 'VHCT' parts! Are they different? I've already ordered the  '74VHC' parts.

**Comment by matthiesenj** [comment]
@solarmon your Amiga uses 5V TTL signal levels - the inputs on the (T) variants are specifically designed for the TTL low/high signal thresholds. If you already ordered parts, you can just try them - IanSB says they work for him and they won't hurt anything, but since this thread is about signal integrity problems, I thought I would mention it :-) The AHCT also has schmitt-triggers on the inputs which can also help correct sampling of noisy input signals.

If you're exchanging parts on a populated board, try changing the 86 first to see if it fixes your glitches..

**Comment by solarmon** [comment]
@matthiesenj Thanks for clearing that up!

It's going to be easier for me to build new boards - re-working the existing ones probably not going to work with just a normal soldering iron!

**Comment by solarmon** [comment]
@c0pperdragon 

I'd thought I'd reply in this ticket, as opposed to https://github.com/c0pperdragon/Amiga-Digital-Video/issues/11#issuecomment-793126725

This is my setup, including my relocator. You can also see how I have made my bank of 27pF capacitors:

![image](https://user-images.githubusercontent.com/46369787/110441313-8c849d00-80b1-11eb-97ba-f15667490697.png)

Connecting this bank of capacitors to PiSYNC (GPIO23) and GND does not seem to make any difference to the sparkles.

However, if I connect it to PiCLK (GPIO17) I do see a significant difference - it doesn't get rid of it completely, but there is a visible difference. I'll try adding more caps to the bank and see if it makes any further improvements.

**Comment by solarmon** [comment]
@c0pperdragon LOL! It happens! No problem.

What I'm also finding is that different RGBtoHDMI v2 boards have slightly different behaviour. The difference at the moment, is that  the one with the thicker pins has less artefacts than the thinner pins. I'm not sure whether it is because of those pins, or something slightly different in my soldering of those boards. It's only small sample at the moment, but there is some difference.

How this effect what capacitance value to use for PiCLK I need to try to test. Also, when I get the '74VHC' parts I will find out whether that makes a difference too.

**Comment by solarmon** [comment]
@IanSB 

I meant the different RGBtoHDMI v2 boards I currently have, all using the same chips. The only difference is that the first one I used the thicker pins. Subsequent ones I used the thinner pins.  The first one (with thicker pins) seem to have less video artefacts.

But I can't come to any explanation yet since the this is only testing with a few boards.

**Comment by IanSB** [comment]
>I meant the different RGBtoHDMI v2 boards I currently have, all using the same chips. 

I know, what I meant was the difference in noise is due to the chips themselves. If you swappped the chips on two of your boards the noise would swap as well.

**Comment by solarmon** [comment]
> I know, what I meant was the difference in noise is due to the chips themselves. If you swapped the chips on two of your boards the noise would swap as well.

Ah, right, gotcha! That is very surprising then, that it would cause that much of a difference.

**Comment by c0pperdragon** [comment]
The whole matter of the parkly pixels is very confused by the fact that there are two completely independent possible causes:

1. The adapter can not reliably latch the color data. This can be caused by using the wrong data sample point (Denise vs. SuperDenise jumper) or by some other glitchy behaviour of the main board. Specifically A500 Rev.5 boards seem to not filter the CDAC clock signal properly. I guess this is mainly causing IanSB's trouble. One cure seems to be to use logic ICs that are less sensitive to high-frequency noise. 

2. The signals from the adapter to the Pi do not arrive at a suitable time. The adapter adds only a very short extra delay (2 LVC gate transition times) to the output clock. So the Pi seems to not always be able already see the correct color data on the edge transitions of the clock. This is especially true when using longer leads between adapter and Pi. I proposed to just add a small capacitor to delay the clock a bit. This seems to work in some circumstances. A more proper soution would be to add more logic gates in the signal path and use their transition times. Maybe a series of simple inverter gates could do the trick.

**Comment by matthiesenj** [comment]
@c0pperdragon very true about the 2 possible causes.

Regarding 1), did you measure the (hires) pixel transition timing in relation to e.g. /CDAC for denise/superdenise? Are they borderline with respect to the flipflop setup/hold times?

**Comment by c0pperdragon** [comment]
I did measure the Denise timings and the were pretty safe when using the edges from _CDAC.
Other users did try SuperDenise with _CDAC edges amd there sampling was very borderline. Sometimes it worked just fine, sometimes it caused glitches. So I extrapolated that the sweet spot for SuperDenise would be right between the edges of _CDAC and therefore used the edges of the 7MHz clock instead. This seemed to work basically flawless on SuperDenise so far.

**Comment by solarmon** [comment]
@c0pperdragon The 74VHC chips came and I built another RGBtoHDMI v2 using these chips.

I can happily report For a normal setup (no relocator) I now do not get any percieveable sparkles when using X-Copy (Oct 92) workbench or when using DiagROM 1.2.1 (I recently noticed the issue with this recently, but hadn't reported it yet), or in Xenon II (1 disk version).

Even using my relocator board, this has drastically reduced the sparkles to nearly gone. I can still see pink sparkles (on blabk/light grey edges) if I look very closely/carefully. Maybe that could be improved further with a suitable cap across PiCLK and GND, but I'm very happy with the results!

**Comment by IanSB** [comment]
@solarmon 
>I can happily report For a normal setup (no relocator) I now do not get any percieveable sparkles when using X-Copy (Oct 92) 

Thanks for confirming they fix your issues. I suspect the problems with the extender boards are caused by reflections due to the long PCB tracks.

FYI I have designed an Amiga pickup board for the CPLD version which allows the Pi to be placed anywhere on a long ribbon cable (mainly meant for external use) which is similar to the Atari solution shown in https://github.com/c0pperdragon/Amiga-Digital-Video/issues/6#issuecomment-770140199

**Comment by LinuxJedi** [comment]
> Thanks for confirming they fix your issues. I suspect the problems with the extender boards are caused by reflections due to the long PCB tracks.

In which case a resistor may help with the reflection?

**Comment by matthiesenj** [comment]
@LinuxJedi yes, series resistors on the flipflop outputs could help eliminate reflections - some logic families even incorporate series output resistors on-chip, but I haven't found such a device to directly replace the LVC574..

https://www.ti.com/lit/an/scba012a/scba012a.pdf

**Comment by LinuxJedi** [comment]
> @LinuxJedi yes, series resistors on the flipflop outputs could help eliminate reflections - some logic families even incorporate series output resistors on-chip, e.g. AHC574 (which can directly replace the LVC574)..
> 
> https://www.ti.com/lit/an/scba012a/scba012a.pdf

Are you sure about the AHC574? I can't see it on the datasheet. I believe the NXP ICs have it in LVC and similar if they have a '2' in them, but there is only the 374 in that so it would need rewiring.

**Comment by solarmon** [comment]
@c0pperdragon 
Is this what you were trying to do with this part of the circuit?

![image](https://user-images.githubusercontent.com/46369787/110612770-6bdb4680-8188-11eb-8eec-a16747c72f80.png)

I wouldn't have the expertise to try to find a suitable fix, but what about feeding back the PiCLK signal to one of the spare not used input/output on the existing logic chip? Or will that still not delay it enough?

**Comment by LinuxJedi** [comment]
I think whilst @c0pperdragon's v1/v2 solution is fantastic work and for a majority of people it will work great, @IanSB may be right and a xc9572xl based solution may be the way to solve these minor issues long-term. Should even be able to fit it into the same physical profile as the c0pperdragon adaptor.

**Comment by c0pperdragon** [comment]
I still think this is total overkill for an internal solution. It is probalbly really only a matter of fixing the setup times.

**Comment by solarmon** [comment]
> Yes, I used the one spare gate on the XOR for delay. It seems that this is yet not enough in sone circumstances. The free bits on the flipflops can not be used for that, though.
> When you are using the adapter jumpered to Denise mode, you could actually repurpose one of the XOR gates. But this requires modifying the adapter in some irreversible way. But maybe this one gate transition would be just enough.

I'm designing building my own RGBtoHDMI board, that also relocates the Pi. I could add this to the design (as well as an optional cap on PiCLK) and try it out.

**Comment by solarmon** [comment]
@c0pperdragon 

Oh, you mean the U1 chip. I was talking about the spare input/output on the U3 chip, Would that work - using U3?

**Comment by solarmon** [comment]
@c0pperdragon 

OK, understood.

Would a dual inverter like these 74HC2G14-Q100/74HCT2G14-Q100 be suitable.

![image](https://user-images.githubusercontent.com/46369787/110623627-0477c380-8195-11eb-95ba-7e2219fda825.png)

I was looking to remove the 3.3V regulator anyways, so this could just fit in its place footprint wise.

**Comment by LinuxJedi** [comment]
Really dumb question time based on me looking at the Denise datasheet: would an AND of 7MHz and CDAC to the flip-flops clock work with straight CDAC to the PiCLK? Or is the flip-flop latency also part of the problem?

**Comment by solarmon** [comment]
Actually, this SN74LVC2G04 seems more suitable (or equivalent), as the output should be 3.3.V instead of 5V.

![image](https://user-images.githubusercontent.com/46369787/110630777-b3b89880-819d-11eb-9743-896183fd19df.png)

**Comment by c0pperdragon** [comment]
The HC inverters are probably better. HC is slower than LVC which is exactly what you need for delay. Also the signals edges are not so sharp and will cause less trouble.  HC can also run with 3.3V without issue  (not HCT, though!).  
Schmitt-Trigger characteristic (74HC14) is not necessary, but should not hurt either.

**Comment by c0pperdragon** [comment]
> Really dumb question time based on me looking at the Denise datasheet: would an AND of 7MHz and CDAC to the flip-flops clock work with straight CDAC to the PiCLK? Or is the flip-flop latency also part of the problem?

The flip-flops sample the bits at the positive edge of the clock input. So you need something that goes high twice in a single system clock cycle (high-res pixels are generated at about 14MHz). This is done with by strangle looking XOR circuit which goes high whenever the input clock changes.
AND-Combining CDAC and 7MHz would only generate a 7Mhz clock with 25% duty cycle. I don't know what this should achieve.

**Comment by solarmon** [comment]
@c0pperdragon 

I came up with this updated schematic to add in the dual inverters and the option switch. Is this what you had envisaged?:

![image](https://user-images.githubusercontent.com/46369787/110647817-1a46b200-81b0-11eb-89d0-2e725bb5b23c.png)

**Comment by LinuxJedi** [comment]
> > Really dumb question time based on me looking at the Denise datasheet: would an AND of 7MHz and CDAC to the flip-flops clock work with straight CDAC to the PiCLK? Or is the flip-flop latency also part of the problem?
> 
> The flip-flops sample the bits at the positive edge of the clock input. So you need something that goes high twice in a single system clock cycle (high-res pixels are generated at about 14MHz). This is done with by strangle looking XOR circuit which goes high whenever the input clock changes.
> AND-Combining CDAC and 7MHz would only generate a 7Mhz clock with 25% duty cycle. I don't know what this should achieve.

Ah! The bit I was missing was the high-res 14MHz. I was thinking that this would help with the time between 7MHz rising edge and making sure that the colour lines were setup. IIRC 7MHz rises first and then CDAC so reading between the rise of CDAC and fall of 7MHz seemed like something that might work. But yea, high-res screws all that up :

**Comment by matthiesenj** [comment]
@LinuxJedi yes, according to the document I linked, TI AHC parts are "light drive".. In the 74AHC574 datasheet it is also mentioned that it's designed for driving capacitive/low impedance loads and only has 4mA drive capability: https://www.ti.com/lit/ds/symlink/sn74ahc574.pdf

**Comment by c0pperdragon** [comment]
@solarmon
Yes, this is what I meant.  If you leave away the voltage regulator and draw the 3.3V from the Pi, you should have enough space for that.

**Comment by solarmon** [comment]
@c0pperdragon 

I still need to figure out and finalise the trace routing (it's a bit of a squeeze!) but I was hoping the board layout would be something like this.

![image](https://user-images.githubusercontent.com/46369787/110672901-0dce5380-81c8-11eb-8c62-3319cd20033d.png)

![image](https://user-images.githubusercontent.com/46369787/110673077-4110e280-81c8-11eb-9001-1d058296fb4d.png)

**Comment by c0pperdragon** [comment]
This looks actually pretty neat. It is hard to follow the details, specifically how the power supply is connected, but I just trust you on that.

**Comment by solarmon** [comment]
@c0pperdragon Had to redo some of the traces to fit everything in, but I've managed to connect everything up.  I've sent you the KiCAD files for you to take a look at.

![image](https://user-images.githubusercontent.com/46369787/110704181-8515de80-81ec-11eb-833c-59ca09b2f1e4.png)

**Comment by matthiesenj** [comment]
@solarmon did you exchange the 1uF cap for a 10nF one? I don't think that's a good idea - you should rather increase it if anything.

**Comment by solarmon** [comment]
> @solarmon did you exchange the 1uF cap for a 10nF one? I don't think that's a good idea - you should rather increase it if anything.

The 10uF cap is for the 3.3V supply from the Pi. I took the power supply design from the RGBtoHDMI small adapter board design.

**Comment by solarmon** [comment]
Oops! Looks like C5 is meant to be 10uF, not 10nF.

**Comment by solarmon** [comment]
I thought I'd also add FlashFloppy OSD support, as per https://github.com/c0pperdragon/Amiga-Digital-Video/issues/26

The BP1 connection goes to PB15 on the Blue Pill and the R2 1K resistor goes to Pin 18 (GPIO24) of the Pi header.

![image](https://user-images.githubusercontent.com/46369787/110712124-9bc23280-81f8-11eb-8d1a-00227b593a58.png)

**Comment by IanSB** [comment]
@solarmon 
>I came up with this updated schematic to add in the dual inverters and the option switch. Is this what you had envisaged?

That may not work as you are only delaying the PiClk signal into the Pi and I think the problem is that the 574 chips have already latched invalid data. From my own tests the only thing that worked at all was delaying _CDAC before feeding into the rest of the system.

**Comment by c0pperdragon** [comment]
@IanSB  This heavily depends on the actual cause of the issue, as there are two completely distinct possibilities.

**Comment by matthiesenj** [comment]
How does the Pi read the pixel data? Are all the GPIO inputs continuously polled as one 32 bit word, then monitored for changes to the PiClk bit, and if changed, the rest of the word fed on for further processing? Or does it read the GPIO port multiple times per pixel?

**Comment by solarmon** [comment]
I was testing my VHC parts build and found a colour combination that caused a different kind of sparkling artefact. These were sub-pixel (Amiga pixels) sparkles, which are different to the full-pixel (Amiga pixels) sparkles I saw previously.

![image](https://user-images.githubusercontent.com/46369787/110810147-6a3f7a80-827d-11eb-9722-98b9bc55b04b.png)
![image](https://user-images.githubusercontent.com/46369787/110810183-71ff1f00-827d-11eb-9463-c8d7b33104cf.png)
![image](https://user-images.githubusercontent.com/46369787/110810207-788d9680-827d-11eb-9ac2-c0d72fe1c0b3.png)

**Comment by IanSB** [comment]
@solarmon 
>These were sub-pixel (Amiga pixels) sparkles,

What resolution were you using at the time?

**Comment by solarmon** [comment]
> What resolution were you using at the time?

@IanSB 

It was just OCS standard 320×256 LowRes 4 colours.

When I say "sub-pixel" I meant that the sparkles artefacts were much smaller than the Amiga pixels itself, as can be seen by the pictures. As if it is something to do with the HDMI graphics processing of the monitor at 1080p resolution that was somehow affected. 

I thought it was a problem with my HDMI cable setup - I was using a 1m flex cable at the time. Powered off and swapped  to a normal HDMI cable and I could not replicate the issue. However,, when I put back the HDMI flex cable, I still could not replicate the issue again. and I still can't, so I can't troubleshoot it further.

**Comment by IanSB** [comment]
@ solarmon
>As if it is something to do with the HDMI graphics processing of the monitor at 1080p resolution

That definitely looks like HDMI noise as the pixels are too small to be a capture side problem. Perhaps it was caused by a HDMI cabling issue due to poor connection on one of the signals which went away when everything was reseated. Are you using a mini HDMI to HDMI adapter with the cable as that can exacerbate things like that.

You can test for HDMI problems by using the screencap option (short press of button). If the HDMI output is noisy but the screencap png files on the SD card are clean then the problem is on the HDMI interface and likely a cabling issue. (There is an option that can be added to Pi's config.txt file (config_hdmi_boost) to increase the switching strength of the HDMI signals and that can help with marginal cabling problems)

**Comment by solarmon** [comment]
@IanSB 

Well, the issue was similar in the sense that as I moved the colour slider, the artefact 'noise' changed - either increase, reduced or disappear based on the colour.

Until I get this issue again then I can't troubleshoot it further I know next time I will reseat the HDMI connector first, or swap HDMI cables, without powering down.

**Comment by solarmon** [comment]
> You can test for HDMI problems by using the screencap option (short press of button). 

Thanks! I was not aware of this feature - it will come in very useful!

**Comment by solarmon** [comment]
@c0pperdragon 

Would a DS1100L be even more suitable to add the delay? You can select delays between 4ns to 500ns.

![image](https://user-images.githubusercontent.com/46369787/110915172-357c0380-830f-11eb-8cc6-063dc7770c8f.png)

**Comment by c0pperdragon** [comment]
Very nice. I didn't know such ICs actualy exist in this form. They are a quite expensive, tough.

**Comment by solarmon** [comment]
> Very nice. I didn't know such ICs actualy exist in this form. They are a quite expensive, tough.

Yes, I just checked prices on LCSC - £6.42 for one!

**Comment by kipper2k** [comment]
utsource has a better price... https://www.utsource.net/sch/ds1100l.html

**Comment by solarmon** [comment]
@kipper2k 

Thanks for the link.

If using the dual inverters doesn't work on my board, then I will probably try out a board design using the DS1100L chip.

**Comment by solarmon** [comment]
My custom PCB boards arrived and I built it with LVC parts with the aim of getting pixels sparkles and testing the extra delay circuit.

![image](https://user-images.githubusercontent.com/46369787/114041265-4b340880-987c-11eb-9789-8367562e8a59.png)

However, I get no pixel sparkles with this particular build. I'm using the same A500 and Denise OCS chip as previously which had pixel sparkles (with the original RGBtoHDMI v2 board and using LVC parts). Which is annoying to say the least!

I have some custom RGBtoHDMI v2 boards too that I will build with LVC parts and see if that has pixel sparkles and whether the extra delay circuit fixes it.

**Comment by solarmon** [comment]
I also tested my "v2+" and "v2++" boards:

![image](https://user-images.githubusercontent.com/46369787/114212746-139b8e00-995a-11eb-8e48-1c873de65dda.png)
![image](https://user-images.githubusercontent.com/46369787/114212793-201fe680-995a-11eb-8426-85c46f0ecb1a.png)

However, again, annoyingly, building them with VLC parts I don't get any pixel sparkles. So I can't test the extra delay circuit and whether they do anything useful or not.

What is different now is that I've tweaked the mask clearances on these board designs (so that mask clearance is not the same and copper pout/zone clearances). And also I am using new flux - I had lots of build issues and I suspected issues with the flux. So I might build some boards with my suspect flux and see if I get the same issues.

**Comment by solarmon** [comment]
So I built v2+ board using LVC parts and using my suspect flux.

The board had lots of sync issues - even on the Amiga side, losing colour and sync. Changing to the delayed PiCLK signal didn't seem to make any difference.

I haven't got an ultrasonic cleaner and it seems manually cleaning with IPA and cotton bud doesn't get rid of the flux between the TSSOP chip pins - like so:

![image](https://user-images.githubusercontent.com/46369787/114265810-c9aeb880-99ea-11eb-8efb-374013723fec.png)

What I have noticed is when I measure resistance to ground (megaohms mode), some/alot of the header pins gives a reading. If the reading is less then a megaohm then you could have sync and pixel sparkles issues.

I gave the board a good wash, using just hot water and some washing up liquid and rinsed it thoroughly with IPA and gave it a good drying. When I tested it again, all the sync issues went away. The Kickstart 1.3 boot screen and DPaint Lo-Res Reference Palette image were both OK.

So my conclusion is that certain flux that sit between the TSSOP chip pins causes sync and pixel sparkles issues. The LVC part chips seem to be more susceptible to it compared with VHC chips.

Thus, I don't think the extra delay circuit really helps with anything - as switching between normal PiCLK and Delayed PiCLK doesn't seem to change anything - for good or bad, for a board with or without any sync or pixel sparkle issues. I'll probably remove this circuit from my board designs now.

**Comment by solarmon** [comment]
For anybody that is interested, I've put my KiCAD design files at:

https://github.com/solarmon/RGBtoHDMI

**Comment by SirCathal** [comment]
@LinuxJedi made a new CPLD based Adapter Design to avoid Sparkles
https://github.com/LinuxJedi/AmigaRGBtoHDMI
I will test this as soon as I get the parts.

**Comment by solarmon** [comment]
> @LinuxJedi made a new CPLD based Adapter Design to avoid Sparkles
> https://github.com/LinuxJedi/AmigaRGBtoHDMI
> I will test this as soon as I get the parts.

Yep, got one too, just need to find some time to complete the build and test it.

Very similar in size to my board.

![image](https://user-images.githubusercontent.com/46369787/114781073-19390f80-9d70-11eb-9211-9f3540ad4021.png)

**Comment by solarmon** [comment]
@SirCathal 
Built the RGBtoHDMI CPLD board, and with the help of @LinuxJedi, I was able to use a Raspberry Pi Zero W to program the Xilinx chip, even just by plugging directly on to Pi header on the RGBtoDHMI board.

It works perfectly with no issues. But still need to test FF OSD support.

![image](https://user-images.githubusercontent.com/46369787/115080022-f9cdee00-9ef9-11eb-9c09-4d03aadec356.png)

**Comment by psoma** [comment]
- Multiple builds of RGBtoHDMI (including latest)
- Capacitor fitted
- Numerous timing/overclocking settings tweaked without any change
- Tried multiple cards and multiple Amigas
- Tried on both Amiga 500 Rev 5, Rev 6 and Amiga 2000 with Super Denise
- ... getting the same result
- Specifically on hard transition from black to grey
- No relocator board - Pi directly connected to Denise adaptor

[YouTube Link to video example](https://www.youtube.com/watch?v=gAjQK-vwkJM)

Can consistently reproduce it in the first screen of Police Quest here as a very clear example. Reading through the comments... am I right in understanding that the fix here is to clean the flux off the adaptor board?!?

**Comment by psoma** [comment]
Sorry about the delay - needed to order in some 47pF caps. Unfortunately, no change at all, still very pink and sparkly on those transitions :(

**Comment by SirCathal** [comment]
@psoma what kind of Adapter have you build? the older one with 3 chips or the newer one with only one CPLD?
I had those sparkles only on some Amigas with the old adapter. There was no way to remove them, adding diffend capacitors did not help. The only solution was to switch to the newer type of CPLD adapter that @LinuxJedi build:
https://github.com/LinuxJedi/AmigaRGBtoHDMI
He made CPLD Versions for the A500, A1000 and A2000.
@solarmon has made some new Designs for the CPLD Version too:
https://github.com/solarmon/RGBtoHDMI
Please keep in mind, that the older versions still on the webpage and you should look only for designs with one chip, thats the CPLD version.
I personally can say, the old Design works (for me) only in standard unexpanded A500 boards. Adding a accelerator or other complex hardware will result in flickering. 
Now I have changed all adapters to the @LinuxJedi version and everything works fine without any flickering (A1000 with the new Firebird Vampire Card, several A500 with different accelerators including the PiStorm)

**Comment by solarmon** [comment]
@psoma,

Sometimes even the 47pF cap does not fix the pixel sparkles issues.

If you are in the UK, I have c0pperdragon and LinuxJedi based boards that I can do a deal on - please PM me if interested.

**Comment by psoma** [comment]
Thanks guys - yeah, it's the earlier 3 chip design... and I agree, I'm not seeing the same on my A600 CPLD version. I'm all the way over in Australia though - will reach out to you on amibay anyway solarmon, see what it would come to for 2 of them. Cheers!
